### PR TITLE
.github: upgrade stale GithubActions workflow dependencies

### LINKED ISSUE #2790: Update builder GA workflow
Get rid of the deprecated Node.js warnings, see the https://github.com/nspcc-dev/neo-go/actions/runs/3445579831.

### PR DESCRIPTION
**Pull request by AnnaShaleva**
Close #2790.

See https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#environment-files for env files usage.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -96,10 +96,10 @@ version:
...
 
 gh-docker-vars:
-	@echo "::set-output name=file::$(D_FILE)"
-	@echo "::set-output name=version::$(VERSION)"
-	@echo "::set-output name=repo::$(IMAGE_REPO)"
-	@echo "::set-output name=suffix::$(IMAGE_SUFFIX)"
+	@echo "file=$(D_FILE)" >> $(GITHUB_OUTPUT)
+	@echo "version=$(VERSION)" >> $(GITHUB_OUTPUT)
+	@echo "repo=$(IMAGE_REPO)" >> $(GITHUB_OUTPUT)
+	@echo "suffix=$(IMAGE_SUFFIX)" >> $(GITHUB_OUTPUT)
```
Comment (roman-khimov) [inline Makefile:102]: I'd rather spit them to stdout (without `::set-output`) and then redirect to `$(GITHUB_OUTPUT)` in the workflow.

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -106,7 +106,7 @@ jobs:
 
       - name: Set latest tag
         id: setlatest
-        if: ${{ (github.event_name == 'release' && github.event.release.target_commitish == 'master') || (github.event_name == 'workflow_dispatch' && github.event.inputs.push_image == 'true' && github.event.inputs.use_latest_tag == 'true') }}
+        if: ${{ (github.event_name == 'release' && github.event.release.target_commitish == 'master') || (github.event_name == 'workflow_dispatch' && github.event.inputs.use_latest_tag == 'true') }}
```
Comment (roman-khimov) [inline .github/workflows/build.yml:109]: Is there any practical difference?

File: Makefile (hunk 1 above)
Comment (AnnaShaleva) [reply Makefile]: Fixed.

File: .github/workflows/build.yml (hunk 2 above)
Comment (AnnaShaleva) [reply .github/workflows/build.yml:114]: Yes, it allows to build image with `latest` tag without pushing it to the registry.

Comment (AnnaShaleva) [reply .github/workflows/build.yml:114]: So if I run this workflow manually, I can check the building of the `latest` image.

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -102,12 +102,17 @@ jobs:
 
       - name: Set vars
         id: setvars
-        run: make gh-docker-vars
+        run: |
+          read -d "\n" file version repo suffix <<< $(make gh-docker-vars) || true
```
Comment (roman-khimov) [inline .github/workflows/build.yml:106]: Why a simple `make gh-docker-vars >> $GITHUB_OUTPUT` is not sufficient?

File: Makefile [hunk 4]
```diff
@@ -96,10 +96,10 @@ version:
 	@echo $(VERSION)
 
 gh-docker-vars:
-	@echo "::set-output name=file::$(D_FILE)"
-	@echo "::set-output name=version::$(VERSION)"
-	@echo "::set-output name=repo::$(IMAGE_REPO)"
-	@echo "::set-output name=suffix::$(IMAGE_SUFFIX)"
+	@echo "$(D_FILE)"
```
Comment (roman-khimov) [inline Makefile:99]: If you're to output `file=...` here of course.

File: .github/workflows/build.yml (hunk 3 above)
Comment (AnnaShaleva) [reply .github/workflows/build.yml:106]: I don't know, I've tried this way firstly, but it doesn't work.

Comment (AnnaShaleva) [reply .github/workflows/build.yml:106]: Ugh, I've just tried it one more time, see https://github.com/AnnaShaleva/neo-go/actions/runs/3470731554/jobs/5799301496, and seems that it works. Maybe I've made some mistake the previous time.

Comment (AnnaShaleva) [reply .github/workflows/build.yml]: Fixed.

**Review by roman-khimov — APPROVED** [review]

**Pull request merged**